utils/cloud_cleanup/gce/clean_gce.py

```diff
@@ -0,0 +1,98 @@
+
+            instance_metadata = {item.key: item.value for item in instance.metadata.items}
+            if should_keep(vm_creation_time, get_keep_hours(instance_metadata)):
+                LOGGER.info("keeping instance %s, keep: %s, creation time: %s ", instance.name,
+                            instance_metadata.get('keep', 'not set'), vm_creation_time)
+                continue
+            if instance_metadata["keep_action"] == 'terminate':
+                if not dry_run:
+                    LOGGER.info("terminating instance %s, creation time: %s", instance.name, vm_creation_time)
+                    try:
+                        res = instances_client.delete(instance=instance.name,
+                                                      project=project_id,
+                                                      zone=instance.zone.split('/')[-1])
+                        res.done()
+                        LOGGER.info("%s terminated", instance.name)
+                    except Exception as exc:  # pylint: disable=broad-except
+                        LOGGER.error("error while terminating instance %s: %s", instance.name, exc)
+                else:
+                    LOGGER.info("dry run: would terminate instance %s, creation time: %s",
+                                instance.name, vm_creation_time)
+
+            elif instance.status == 'RUNNING':  # stop the rest, only running instances
+                if not dry_run:
+                    LOGGER.info("stopping instance %s, creation time: %s", instance.name, vm_creation_time)
+                    try:
+                        res = instances_client.stop(instance=instance.name,
+                                                    project=project_id,
+                                                    zone=instance.zone.split('/')[-1])
+                        res.done()
+                        LOGGER.info("%s stopped", instance.name)
+                    except Exception as exc:  # pylint: disable=broad-except
+                        LOGGER.error("error while stopping instance %s: %s", instance.name, exc)
+                else:
+                    LOGGER.info("dry run: would stop instance %s, creation time: %s", instance.name, vm_creation_time)
+
+
+if __name__ == "__main__":
+    arg_parser = argparse.ArgumentParser('gce_cleanup')
+    arg_parser.add_argument("--duration", type=int,
+                            help="duration to keep non-tagged instances running in hours",
+                            default=os.environ.get('DURATION', DEFAULT_KEEP_HOURS))
+    arg_parser.add_argument("--dry-run", action=argparse.BooleanOptionalAction,
+                            help="do not stop or terminate anything",
+                            default=os.environ.get('DRY_RUN', False))
+
+    args = arg_parser.parse_args()
+
+    is_dry_run = bool(args.dry_run)
+
+    if is_dry_run:
+        LOGGER.error("'Dry run' mode on")
+
+    projects = ['sct-project-1', 'local-ssd-latency']
```

Comment:
we have one project missing here, we should use `SUPPORTED_PROJECTS`

```
SUPPORTED_PROJECTS = {'gcp', 'gcp-sct-project-1',
                      'gcp-local-ssd-latency'} | {os.environ.get('SCT_GCE_PROJECT', 'gcp-sct-project-1')}
```

Reply:
I don't understand it - we don't have project that are prefixed with `gcp-` - I believe it is added just for downloading credentials and we take project from credentials info (please correct me if I'm wrong). So we actually have 2 projects 'sct-project-1` and `local-ssd-latency`. I don't know then what is `gcp` project in that case.

Somehow, default credentials work for both of them.

Reply:
We have 3 projects 

skilled-adapter-452 is what we call gcp (it was the first project.

We still want to scan it 

You can take a look at the emails report, people are running machines there, some from out older branches from before we used sct-project-1

Reply:
again your'e right, and I got into problem of insufficient privileges - so I adapted code to work similarly as in other places (creating a new client for each credentials and reusing credential's project_id).